src/comp_loinc/analysis/subclass_rels.py

```diff
@@ -14,65 +14,117 @@
 import logging
 from argparse import ArgumentParser
 from pathlib import Path
-from typing import Dict, Set, Tuple, Union
+from typing import Dict, List, Optional, Set, Tuple, Union
 
 import matplotlib.pyplot as plt
 import pandas as pd
 from jinja2 import Template
 from tabulate import tabulate
 from upsetplot import from_contents, UpSet
 
-logger = logging.getLogger(__name__)
-
 from comp_loinc.analysis.utils import bundle_inpaths_and_update_abs_paths, cli_add_inpath_args, \
-    _subclass_axioms_and_totals
+    _subclass_axioms_and_totals, CLASS_TYPES, _filter_classes, _disaggregate_classes_from_class_list
 
 THIS_DIR = Path(os.path.abspath(os.path.dirname(__file__)))
 PROJECT_ROOT = THIS_DIR.parent.parent.parent
 MISSING_AXIOMS_PATH = PROJECT_ROOT / "output" / "tmp" / "missing_comploinc_axioms.tsv"
 DESC = "Analysis for totals and overlap of subclass axioms / relationships between LOINC, CompLOINC, and LOINC-SNOMED."
 DEFAULTS = {
     'loinc-path': 'output/tmp/subclass-rels-loinc-indirect-sc-axioms-included.tsv',
-    'loinc-snomed-path': 'subclass-rels-loinc-snomed-indirect-sc-axioms-included.tsv',
-    'comploinc-primary-path': 'output/tmp/subclass-rels-comploinc-indirect-included-primary.tsv',
-    'comploinc-supplementary-path': 'output/tmp/subclass-rels-comploinc-indirect-included-supplementary.tsv',
-    'outpath-md': 'documentation/analyses/class-depth/depth.md',
-    'outpath-upset-plot': 'documentation/analyses/class-depth',
+    'loinc-snomed-path': 'output/tmp/subclass-rels-loinc-snomed-indirect-sc-axioms-included.tsv',
+    'comploinc-all-primary-path': \
+'output/tmp/subclass-rels-comploinc-indirect-included-all-primary.tsv',
+    'comploinc-all-supplementary-path': \
+'output/tmp/subclass-rels-comploinc-indirect-included-all-supplementary.tsv',
+    'comploinc-LOINC-primary-path': \
+'output/tmp/subclass-rels-comploinc-indirect-included-LOINC-primary.tsv',
+    'comploinc-LOINC-supplementary-path': \
+'output/tmp/subclass-rels-comploinc-indirect-included-LOINC-supplementary.tsv',
+    'comploinc-LOINCSNOMED-primary-path': \
+        'output/tmp/subclass-rels-comploinc-indirect-included-LOINCSNOMED-primary.tsv',
+    'comploinc-LOINCSNOMED-supplementary-path': \
+        'output/tmp/subclass-rels-comploinc-indirect-included-LOINCSNOMED-supplementary.tsv',
+    'outpath-md': 'documentation/subclass-analysis.md',
+    'outdir-upset-plots': 'documentation',
+    'class-types': ('terms',),
+}
+VARIATIONS: Dict[str, List[str]] = {
```

Comment:
This controls upset plot generation:
- Their title
- The sources / flavors that comprise them